### DISCUSSION TITLE
Renovate can't find local docker dependency

### DISCUSSION THREAD

**Discussion by cogito-cmurphy** [renovatebot/renovate#10057] [Request Help] [answered]
Hello,

I am attempting to run renovate against a single Dockerfile. Below is the docker file. Renovate cannot look up the dependency 
`FROM docker.cogitocorp.us/envoy-preflight` . I believe I have my artifactory credentials setup properly so that it should be able to be looked up, ( I am also sharing the credentials via IPS incase the code snippet below isn't correct.).  
`environment:
          RENOVATE_HOST_RULES=\''[{"hostType":"docker","domainName":"docker.cogitocorp.us","username":"'"$RENOVATE_DOCKER_USER"'"',"password":"'"$RENOVATE_DOCKER_PASS"'"}]'\'`
```
#  envoy-preflight
#

FROM docker.cogitocorp.us/envoy-preflight:1.0-cogito-1 AS preflight

#
# cogito-base
#

FROM debian:buster

# Add preflight and set ENTRYPOINT
COPY --from=preflight /envoy-preflight /envoy-preflight
ENV ENVOY_ADMIN_API http://127.0.0.1:15000

# Install core rpms
RUN apt-get update && apt-get install -y \
  git \
  curl \
  jq \
  sudo \
  tree \
  unzip \
  vim \
  wget \
  zip
```
Here is the warning I receive from Renovate. 
```
       "config": {
         "dockerfile": [
           {
             "packageFile": "v2/cogito-base/21.02.24/Dockerfile",
             "deps": [
               {
                 "depName": "docker.cogitocorp.us/envoy-preflight",
                 "currentValue": "1.0-cogito-1",
                 "replaceString": "docker.cogitocorp.us/envoy-preflight:1.0-cogito-1",
                 "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
                 "datasource": "docker",
                 "depType": "stage",
                 "depIndex": 0,
                 "updates": [],
                 "warnings": [
                   {
                     "topic": "docker.cogitocorp.us/envoy-preflight",
                     "message": "Failed to look up dependency docker.cogitocorp.us/envoy-preflight"
                   }
                 ]
```

Any assistance in getting this to be looked up would be greatly appreciated.

Thank you,
Connor

**Comment by rarkins** [comment, marked as answer]
The env you past in doesn't look right - or at least has an unexpected number of `'` within the JSON, which is not an allowed character. Check the log at the very start of Renovate to see the processing of the config.

**Reply by cogito-cmurphy** [reply]
Thank you for the quick response. I have removed that env variable for now to see if there were any changes. It still processes the same way. I believe because I am passing the artifactory credentials via IPS (imagePullSecrets) that the bad env wouldn't impact anything.

I am a bit confused on how renovate looks at dependencies. 

We have
`docker.cogitocorp.us/envoy-preflight:1.0-cogito-1`

Which is an internal docker image on our private repo : tagged. When renovate goes to look for the dependency it looks for `docker.cogitocorp.us/envoy-preflight`. When I look in Artifactory it looks like it should be looking up `docker/envoy-preflight/` which is what the repo path is. Is there logic in renovate to support this? Should we maybe move to git tags?

Thanks,
Connor

**Reply by viceice** [reply]
i think you did something wrong, the name in the dockerfile must match the registry image name.

Docker ( and renovate will lookup `docker.cogitocorp.us/envoy-preflight` as `https://docker.cogitocorp.us/v2/envoy-preflight/tags/list`

so if you need `https://docker.cogitocorp.us/v2/docker/envoy-preflight/tags/list`, the image name needs to be `docker.cogitocorp.us/docker/envoy-preflight`.

This is official docker registry api https://docs.docker.com/registry/spec/api/#tags

**Reply by viceice** [reply]
if you like to update from `1.0-cogito-1` to `1.0-cogito-2` you need to use regex or loose versioning.

but loose won't work from `1.0-cogito-9` to `1.0-cogito-10`, as it only compares by string.

**Reply by cogito-cmurphy** [reply]
@viceice interesting, ill take a look at loose versioning. I will also ask the team about the /tags/list. I can't seem to find that URL for our local artifactory but it might be a me thing. Thanks for the assistance.

**Reply by cogito-cmurphy** [reply]
I see now whats wrong, renovate isnt getting the credentials needed to authenticate. When I do a local lookup with credentials I can see a tag list (at the proper url, you listed above). How can I pass an encrypted password/ASM/env var to Renovate so that it can authenticate?

hostRules seems to be a json blob which won't take env variables.

Thank you

**Reply by rarkins** [reply]
hostRules can be configured in env variables. e.g. run `node`, `const rules = [....]` and then `JSON.stringify(rules)`. The value you get - which should be wrapped in single quotation marks - can be added to env as `RENOVATE_HOST_RULES`.

**Reply by cogito-cmurphy** [reply]
thank you @rarkins  I am going to close this question as it has been answered.

**Reply by viceice** [reply]
checkout this: https://github.com/renovatebot/renovate/discussions/10072#discussioncomment-762851

**Reply by cogito-cmurphy** [reply]
thanks @viceice  im going to try that
